core/simulations/index.ts

```diff
@@ -0,0 +1,44 @@
+import prompts, { PromptObject } from 'prompts';
+import { SIMULATIONS } from './config';
+import readline from 'readline';
+
+const keypress = async () => {
+    await prompts({
+        type: 'invisible',
+        name: 'irrelevant',
+        message: 'Press Enter to continue',
+    });
+};
+
+const selectAndRunSimulation = async () => {
+    readline.emitKeypressEvents(process.stdin);
+    const promtChoices = SIMULATIONS.map(simulationConfig => simulationConfig.title).map(simulationName => ({
+        title: simulationName,
+        value: simulationName,
+    }));
+    const promtConfig: PromptObject = {
+        type: 'select',
+        name: 'value',
+        message: 'Select Simulation',
+        choices: promtChoices,
+    };
+    while (true) {
```

Comment:
Maybe the loop should start on the top of the file, so that when one simulation is completed, the other can be selected?

Reply:
i don't understand nor the suggestion, nor the reasoning here:

  - "top of the file" start of the loop does not make sense to me: what about imports and helper function definitions?
    - if "top of the function" is actually meant, then effectively this is already the case because higher we only have the variable definitions that are the same for every loop
  - the other simulation can already be selected after the previous one is concluded.

Reply:
Alright, I got confused by the generic `const answer = await prompts(promtConfig)` which doesn't really say what it does. This makes code more readable:

```ts
const selectedSimation = await prompts(selectSimulationPrompt);
const simulationConfig = SIMULATIONS.find(simulation => simulation.title === selectedSimation.value);
```